tests/ui/perfherder/graphs-view/test_data_modal_test.jsx

```diff
@@ -0,0 +1,204 @@
+  return render(
+    <TestDataModal
+      projects={repos}
+      plottedUnits={plottedUnits}
+      timeRange={{ value: 172800, text: 'Last two days' }}
+      getTestData={() => {}}
+      options={{}}
+      testData={[]}
+      frameworks={frameworks}
+      showModal
+      toggle={() => {}}
+      getInitialData={() => ({
+        platforms,
+      })}
+      getSeriesData={mockGetSeriesData}
+    />,
+  );
+};
+
+afterEach(cleanup);
+
+test('Tags multi select is not shown if series data has no tags', async () => {
+  const { queryByLabelText } = testDataModel();
+
+  expect(queryByLabelText('Tags')).toBeNull();
+});
+
+test('Tags multi select is shown if series data has tags', async () => {
+  const { getByLabelText } = testDataModel();
+  const multiSelect = await waitFor(() => getByLabelText('Tags'));
+
+  expect(multiSelect).toBeInTheDocument();
+});
+
+test('Tag used in multiple tests appears once in the tags list', async () => {
+  const { getAllByTestId } = testDataModel();
+  const tags = flatMap(seriesDataWithRepeatedTag, (data) => data.tags);
+  const tagOccurency = tags.filter((tag) => tag === 'repeated-tag').length;
+
+  expect(tagOccurency).toBeGreaterThan(1);
+
+  const tag = await waitFor(() => getAllByTestId('available-tag repeated-tag'));
+
+  expect(tag).toHaveLength(1);
+});
+
+test('Selecting a tag from tags multi select shows the tests that have the specific tag', async () => {
+  const { queryByTestId, getByText, getByTestId } = testDataModel();
+  const tag = await waitFor(() => getByText(seriesData[0].tags[0]));
+
+  fireEvent.click(tag);
+
+  const activeTag = await waitFor(() =>
+    getByTestId(`active-tag ${seriesData[0].tags[0]}`),
+  );
+
+  expect(activeTag).toBeInTheDocument();
+
+  const tests = await waitFor(() => queryByTestId('tests'));
+  const test = await waitFor(() => queryByTestId(seriesData[0].id.toString()));
+
+  expect(tests).toContain(test);
+});
+
+test("Selecting 'all tags' from tags multi select shows only the tests that have tags", async () => {
+  const { getByTestId, getByText } = testDataModel();
+  const testsWithTags = seriesData.filter((test) => test.tags.length);
+
+  const tag = await waitFor(() => getByText('all tags'));
+
+  fireEvent.click(tag);
+
+  const tests = await waitFor(() => getByTestId('tests'));
+
+  expect(tests).toHaveLength(testsWithTags.length);
+});
+
+test("Selecting 'all tags' from tags multi select makes all the available tags active", async () => {
+  const { getAllByTestId, getByText } = testDataModel();
+  const availableTags = await waitFor(() => getAllByTestId(/available-tag/));
+  const allTags = await waitFor(() => getByText('all tags'));
+
+  fireEvent.click(allTags);
+
+  const activeTags = await waitFor(() => getAllByTestId(/active-tag/));
+
+  expect(activeTags).toHaveLength(availableTags.length - 1);
+});
+
+test('Selecting two tags from tags multi select show the tests that have at least one of those tags', async () => {
```

Comment:
The old filter text-based implementation used to AND the tags. This one seems to OR them.
We should stick to the old implementation.

Users typically use more & more tags just to narrow down the list, not to extend it.